internal/controller/streaming_manager.go

```diff
@@ -85,6 +99,8 @@ func (sm *streamManager) BootUpStreamAndReconnect(ctx context.Context) error {
 	}
 
 	snapshotCompleted.Add(1)
+	dd.timeStarted = time.Now()
```

Comment:
`Lock` the `Mutex` while modifying those fields. Otherwise, other goroutines, e.g. the healthz handler won't be able to read those updates.